protocol/messenger_handler.go

```diff
@@ -406,20 +412,114 @@ func (m *Messenger) handleCommandMessage(state *ReceivedMessageState, message *c
 	return nil
 }
 
-func (m *Messenger) HandleBackup(state *ReceivedMessageState, message protobuf.Backup) error {
+func (m *Messenger) HandleBackup(state *ReceivedMessageState, message protobuf.Backup) []error {
+	var errors []error
+	err := m.handleSyncProfile(state, message.Profile, message.Clock)
+	if err != nil {
+		errors = append(errors, err)
+	}
+
 	for _, contact := range message.Contacts {
-		err := m.HandleSyncInstallationContact(state, *contact)
+		err = m.HandleSyncInstallationContact(state, *contact)
 		if err != nil {
-			return err
+			errors = append(errors, err)
 		}
 	}
 
 	for _, community := range message.Communities {
-		err := m.handleSyncCommunity(state, *community)
+		err = m.handleSyncCommunity(state, *community)
 		if err != nil {
-			return err
+			errors = append(errors, err)
 		}
 	}
+	err = m.handleSyncSettings(state, message.Settings)
+	if err != nil {
+		errors = append(errors, err)
+	}
+
+	if m.config.messengerSignalsHandler != nil {
+		response := wakusync.SyncFromWakuResponse{}
+
+		response.AddWakuSyncDetails(SyncWakuSectionKeyProfile, message.ProfileDetails)
+		response.AddWakuSyncDetails(SyncWakuSectionKeyContacts, message.ContactsDetails)
+		response.AddWakuSyncDetails(SyncWakuSectionKeyCommunities, message.CommunitiesDetails)
+		response.AddWakuSyncDetails(SyncWakuSectionKeySettings, message.SettingsDetails)
+
+		m.config.messengerSignalsHandler.SendSyncFromWakuProgress(&response)
+	}
+
+	return errors
+}
+
+func (m *Messenger) handleSyncProfile(state *ReceivedMessageState, message *protobuf.SyncProfile, backupTime uint64) error {
+	if message == nil {
+		return nil
+	}
+
+	pubKey := common.PubkeyToHex(&m.identity.PublicKey)
+	contact, ok := state.AllContacts.Load(pubKey)
+	if !ok {
+		return errors.New("cannot find logged in profile among all known contacts")
+	}
+
+	response := wakusync.SyncFromWakuResponse{
+		Profile: &wakusync.SyncProfile{},
+	}
+
+	if contact.LastUpdated < backupTime {
+		err := m.SetDisplayName(message.DisplayName, false)
+		response.AddDisplayName(message.DisplayName, err == nil)
+	}
+
+	syncWithBackedUpImages := false
+	dbImages, err := m.multiAccounts.GetIdentityImages(message.KeyUid)
+	if err != nil {
+		syncWithBackedUpImages = true
+	}
+	if len(dbImages) == 0 {
+		if len(message.Pictures) > 0 {
+			syncWithBackedUpImages = true
+		}
+	} else {
+		// since both images (large and thumbnail) are always stored in the same time, we're free to use either of those two clocks for comparison
+		lastImageStoredClock := dbImages[0].Clock
+		syncWithBackedUpImages = lastImageStoredClock < backupTime
+	}
+
+	if syncWithBackedUpImages {
+		if len(message.Pictures) == 0 {
+			err = m.multiAccounts.DeleteIdentityImage(message.KeyUid)
+			response.AddImages(nil, err == nil)
```

Comment:
This looks a bit sketchy, what is it trying to do exactly?

Reply:
There are 2 properties describing each change (`AddImages`, `AddDisplayName` they all expect result of real action did the real go well or not)... cause client needs to know about the action is it completed or not, in some cases we may want to notify user about what's really happened.

For example in case of setting new identity images, we're sending a list of images as `images` property and boolean flag as `imagesStored` property, then on the client side we always check `if imagesStored` and then do the things, if not we may notify user what went wrong and offer him to store images again.

So a short answer is `err == nil` notifies client if the action went well or not.

Reply:
This is a very uncustomary  way to handle errors, I think it's best to understand why something would fail, if it's expected, we should handle it, if it's not expected, we should fail fast, this seems like it's really swallowing the error and continuing, leaving it up to the client to handle something that's not able to handle.
Are there any expected errors happening here? or you don't know how it could fail?

Reply:
> Are there any expected errors happening here? or you don't know how it could fail?

At this point errors which may happen are only those returned from `db.Exec` call. 

Now for example if error happened for setting display name action we will send a signal containing `{"CustomName", false}` if there is no errors we send `{"CustomName", true}`, when client receives such signal, it will firstly check the value and if action was performed with success or not and do the things accordingly. For example if all went well it will update display where is needed, if didn't go well it will display an error to a user or offer him to try to store `"CustomName"` again. That was my idea when I was working on this, always to have an option.

Reply:
I think in this case there's nothing we can really do, if the database is not responsive, we might want to bubble up an error to the user, but we should stop immediately any operation, there's no easy way to gracefully handle this